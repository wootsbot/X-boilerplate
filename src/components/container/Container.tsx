"use client";

import styles from "./Container.module.css";

function Container({ ...others }) {
  return <div className={styles.root} {...others} />;
}

export default Container;
