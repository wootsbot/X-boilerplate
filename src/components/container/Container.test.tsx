import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import Container from "./Container";

describe("<Container />", () => {
  test("must render correctly", () => {
    const { container } = render(
      <Container>
        <h1>Container</h1>
      </Container>,
    );

    expect(
      screen.getByRole("heading", {
        name: /container/i,
      }),
    ).toBeDefined();
    expect(screen.getByRole("heading")).toBeDefined();

    expect(container).toMatchSnapshot();
  });
});
