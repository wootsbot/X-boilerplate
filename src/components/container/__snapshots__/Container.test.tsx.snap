// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Container /> > must render correctly 1`] = `
<div>
  <div
    class="_root_448038"
  >
    <h1>
      Container
    </h1>
  </div>
</div>
`;
